File: Data Structures & Algorithms/reverse-a-linked-list/submission-1.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public ListNode reverseList(ListNode head) {
        ArrayList<Integer> l=new ArrayList<>();
        ListNode p=head;
        while(p!=null){
            l.add(p.val);
            p=p.next;
        }
        Collections.reverse(l);
        p=head;
        int i=0;
        while(p!=null){
            p.val=l.get(i);
            i+=1;
            p=p.next;
        }
        return head;
    }
}

File: Data Structures & Algorithms/rotate-list/submission-1.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def rotateRight(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        l=[]
        cur=head
        if head==None:
            return head
        while cur:
            l.append(cur.val)
            cur=cur.next
        k%=len(l)
        for i in range(k):
            j=l.pop()
            l.insert(0,j)
        i=0
        cur=head
        while cur:
            cur.val=l[i]
            i+=1
            cur=cur.next
        return head

        